opslevel/resource_opslevel_integration_google_cloud.go

```diff
@@ -0,0 +1,277 @@
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"installed_at": schema.StringAttribute{
+				Description: "The time that this integration was successfully installed, if null, this indicates the integration was not completely installed.",
+				Computed:    true,
+			},
+			"name": schema.StringAttribute{
+				Description: "The name of the integration.",
+				Required:    true,
+			},
+			"ownership_tag_keys": schema.SetAttribute{
+				ElementType: types.StringType,
+				Description: "An Array of tag keys used to associate ownership from an integration. Max 5",
+				Required:    true,
+				Validators: []validator.Set{
+					setvalidator.SizeAtMost(5),
+				},
+			},
+			"ownership_tag_overrides": schema.BoolAttribute{
+				Description: "Allow tags imported from Google Cloud to override ownership set in OpsLevel directly.",
+				Required:    true,
+			},
+			"projects": schema.ListAttribute{
+				Description: "A list of the Google Cloud projects that were imported by the integration.",
+				Computed:    true,
+				ElementType: types.ObjectType{
+					AttrTypes: googleCloudProjectAttrs(),
+				},
+			},
+		},
+	}
+}
+
+func (r *integrationGoogleCloudResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var planModel integrationGoogleCloudResourceModel
+
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &planModel)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	ownershipTagKeys, diags := SetValueToStringSlice(ctx, planModel.OwnershipTagKeys)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	input := opslevel.GoogleCloudIntegrationInput{
+		ClientEmail:           planModel.ClientEmail.ValueStringPointer(),
+		Name:                  planModel.Name.ValueStringPointer(),
+		OwnershipTagKeys:      &ownershipTagKeys,
+		PrivateKey:            planModel.PrivateKey.ValueStringPointer(),
+		TagsOverrideOwnership: planModel.TagsOverrideOwnership.ValueBoolPointer(),
+	}
+
+	createdIntegration, err := r.client.CreateIntegrationGCP(input)
+	if err != nil {
+		resp.Diagnostics.AddError("opslevel client error", fmt.Sprintf("Unable to create Google Cloud integration, got error: '%s'", err))
+		return
+	}
+
+	stateModel := newIntegrationGoogleCloudResourceModel(ctx, *createdIntegration, planModel, &resp.Diagnostics)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	tflog.Trace(ctx, "created a Google Cloud integration")
+	resp.Diagnostics.Append(resp.State.Set(ctx, &stateModel)...)
+}
+
+func (r *integrationGoogleCloudResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var stateModel integrationGoogleCloudResourceModel
+
+	resp.Diagnostics.Append(req.State.Get(ctx, &stateModel)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	readIntegration, err := r.client.GetIntegration(opslevel.ID(stateModel.Id.ValueString()))
```

Comment:
```suggestion
	readIntegration, err := r.client.GetIntegration(asID(stateModel.Id))
```
to make things a little simpler

Reply:
Corrected